### ISSUE TITLE
Document use of all supported serializers in R documentation

### ISSUE THREAD

**Issue by gothub** [ropensci/redland-bindings#52]
Currently the R documentation (R help, vignettes) describes the rdfxml serializer only.
Update documentation to describe all other available formats (turtle, ntriples, ...), complete
with examples for each format.

**Comment by gothub** [comment]
Supported raptor parsers and the names to use to access these are documented in http://librdf.org/raptor/api/raptor-formats-types-by-parser.html

**Comment by cboettig** [comment]
It would be really nice if the R package were able to select the appropriate mime type from the table if the parser name was given, but maybe that's beyond the scope of redland-bindings.  

I've been wondering about having a higher-level interface that could streamline the commands required for some common tasks by wrapping the redland functions up with some sensible defaults.  e.g. so one could do `convert(input_file, input_format, output_format)`, or `query(sparql_query, file, format)`.  Looking for some good example models for a simple interface.  `rdflib` in python seems to have a pretty compelling simple interface: https://rdflib.readthedocs.io/en/latest/intro_to_sparql.html

**Comment by mbjones** [comment]
That sounds like a great idea, @cboettig.  Lots of converters can figure it out based on filename extensions too.  I could see a converter that defaulted the input and output formats based on extension if the format arguments were missing, so one might use `rdf_convert(input_file, output_file, input_format=NA, output_format=NA)`, which might be used as:

```R
rdf_convert('model.xml', 'model.ttl')
rdf_convert('model.ttl', 'model.n4', output_format='nquads')
rdf_convert('model.ttl', 'model.n4', input_format='turtle', output_format='nquads')
```

**Comment by cboettig** [comment]
@mbjones indeed.  Looked like there might be some utility within redland to guess the file type (maybe by looking at the actual syntax) which would be better than the extension, but I couldn't find it.  (There was some documentation of `guess` as a parser type, but I couldn't get that to do anything).  

Using extensions sounds nice and simple. I'm actually not familiar with the most common extensions for each of the rdf serializations, is there a good table of that somewhere handy?

For kicks, I've put together a light wrapper around the common functions, modeling a bit off of rdflib interface in python: https://github.com/cboettig/rdflib  

Once it's a bit more mature, happy to roll that into redland if that makes sense, or keep it as a separate wrapper or whatnot.  Obviously could support more config options (e.g. passing additional arguments to the `Storage` class, once I get my head around what those actually do), but for now it's mostly a convenience for me to save typing.

**Comment by gothub** [comment]
Include documentation of the recommended formats, with examples, in the `Parser` and `Serializer` classes as well as vignettes,

**Comment by gothub** [comment]
Regarding the specifying of a parser name and MIME type combination as detailed in
http://librdf.org/raptor/api/raptor-formats-types-by-parser.html, it appears that only the
first MIME type listed for a parser name works. The raptor2-2.0.15 source code indicates that
these are the 'default' MIME types for a parser/serializer name. I'm assuming that this is a bug in 
either the redland, or raptor library, in which only the default MIME type can be used. Note that 
the MIME type can be omitted in the R call and the parser/serializer is found:

`parser <- new("Parser", world, name = 'rdfxml', mimeType = '')'

This call creates a functioning parser.

I'll inspect the redland/raptor source a bit more to see if I can find exactly where the problem is, but for now I'm assuming it's the above mentioned bug where only the default MIME type can be
specified.

**Comment by cboettig** [comment]
@gothub my impression was that only the `q1.0` MIME types worked(?).  Not exactly sure what the significance of the `q` numbers is, but seems to imply older versions? Or can you get the older `q` types to work (at least in cases where there is no `q1.0` like with grddl or rss?

**Comment by gothub** [comment]
After a bit of searching about, I found a reference to the 'q' values here:

https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

So the 'q' value is a parameter to `HTTP Accept` headers that is used to specify the relative
quality value' acceptable for a MIME type selected during content negotiation. The q values can range  from q=0.0 to q=1.0 (0=no preference, 1.0=prefer over all other formats).

Not sure where the redland developers got the q values specified in their documentation, maybe
it's a geeky way of stating what they think the preferred MIME-type should be for a file type
and related parser. In any case, it makes sense that the q=1.0 MIME types work, and that they
are ones designated as default in the raptor source code, for example, for the turtle parser
from the raptor2-2.0.15/src/turtle_parser.c:

```
static const raptor_type_q turtle_types[TURTLE_TYPES_COUNT + 1] = {
  /* first one is the default */
  { "text/turtle", 11, 10},
  { "application/x-turtle", 20, 10},
  { "application/turtle", 18, 10},
  { "text/n3", 7, 3},
  { "text/rdf+n3", 11, 3},
  { "application/rdf+n3", 18, 3},
  { NULL, 0}
};
```

So, this all leads me to believe that not being able to specify mime-types other than the
first one is a bug.

**Comment by cboettig** [comment]
Okay, with a little more experimentation I found that `text/turtle` works, but the other `turtle` mime types are not working (despite all being `q 1.0`). 

Also, I cannot get `trig` to work with the `q 1.0` mime type or any other obvious permutation.  

Working so far for me are: 'rdfxml', 'turtle', 'nquads', 'ntriples'.   Haven't tested `rdfa` or `rss`. Is `n3` the same thing as `turtle`?
